src/main/java/redis/clients/jedis/JedisPool.java

```diff
@@ -392,4 +395,39 @@ public void returnResource(final Jedis resource) {
       }
     }
   }
+
+  public void withJedisDo(Consumer<Jedis> consumer) {
+    try (Jedis jedis = getResource()) {
+      consumer.accept(jedis);
+    }
+  }
+
+  public <K> K withJedisGet(Function<Jedis, K> function) {
+    try (Jedis jedis = getResource()) {
+      return function.apply(jedis);
+    }
+  }
+
+  ConnectionProvider asConnectionProvider(){
+    return new JedisPoolConnectionProvider();
+  }
+
+  private class JedisPoolConnectionProvider implements ConnectionProvider {
+
+    @Override
+    public Connection getConnection() {
+      return JedisPool.this.getResource().getConnection();
```

Comment:
You're loosing the reference of the Jedis object; so you can't call `close()` on it and so it can't be returned to the pool.
The pool will be exhausted in a multi-threaded environment too quickly.

Reply:
Yep, I checked with a test. 
I'm trying to fix around, but it seems impossible.
I'll try but if I don't find a way, I will close and discard this PR

Reply:
Yep, after reviewing the code and trying to make it work in local, I see there's no use (for now)

I wanted to to this because my library uses JedisPool and anyone using JedisPooled in a project could have problems integrating it; so for a future release migth be a good idea. 

But now better close this PR